### ISSUE TITLE
Config noImplicitAny seems to be not working since typescript 2.1

### ISSUE THREAD

**Issue by Flarna** [Microsoft/TypeScript#18679]
**TypeScript Version:**  2.5.2 / nightly (2.6.0-dev.20170922)

**Code**
tsconfig.json with `noImplicitAny` set to true:
```json
{
  "compilerOptions": {
    "target": "es5",
    "module": "commonjs",
    "strict": true,
    "noImplicitAny": true
  }
}
```
Creating a variable without specifying a type:
```ts
let i;
i = 5;
i = "foo";
```

**Expected behavior:**
Typescript compiler complains that `i` is implicit any.

**Actual behavior:**
Typescript compile does not report any issue.

If I use typescript 2.0.10 (with adapted tsconfig.json) I get "error TS7005: Variable 'i' implicitly has an 'any' type." as expected.

**Comment by HerringtonDarkholme** [comment]
Actually it is a new feature.

https://github.com/Microsoft/TypeScript/pull/11263

**Comment by Flarna** [comment]
Ah ok thanks, thats fine.
Seems I got confused by VsCode which shows `let i: any` if I hover my mouse pointer over the statement `i = 5`.
